### ISSUE TITLE
Feature Request: Implement password protection for `models.py` & `piechart.py`

### ISSUE THREAD

**Issue by this-is-yaash** [AmateursLeague/sneaky-package#14]
A new feature that allows users to store files in the package in an encrypted format. When a user attempts to open one of these files, the application should prompt for a password to decrypt and access the file's contents.

**Comment by Varun-malviya5** [comment]
Hi , i would like to work on this feature . Please assign me this.
My approach is as follows:
1. Using the required python libraries
2. Encrypting the files and using the library to make its unique key
3. By using the input command can prompt the user to enter the passcode always
4. Further will  handle the edge cases

**Comment by this-is-yaash** [comment]
Can you implement this without encryption?  
So far, we have 3 features, each with its own methods:
1. `models.py` - Displays the output.
2. `graph.py` - Copies it to the system's clipboard.
3. `piechart.py` - Writes the file.

Can you make it so that the program executes properly only if a password is provided?

I want the password to be dynamic, combining it with the time or date. For example, entering the current time should execute the program.

I’m not sure, but I think this approach would be better than using encryption.

**In simple terms,**  
Instead of encryption, implement regular password protection. The password should change every minute, based on the current time.

**Example:** If the time is 12:00:
```python
from package_name.models import display
display('filename', 1200)
```
This should execute the feature, with the time serving as the password.

For this issue, you can focus on the `models.py` feature only.  
We’ll create two additional issues for implementing the other features, as they have different functions—one copies to the clipboard and the other writes the entire file to the root directory.

**Comment by sid7219** [comment]
can you assign this issue to me

**Comment by sid7219** [comment]
> Can you implement this without encryption? So far, we have 3 features, each with its own methods:
> 
> 1. `models.py` - Displays the output.
> 2. `graph.py` - Copies it to the system's clipboard.
> 3. `piechart.py` - Writes the file.
> 
> Can you make it so that the program executes properly only if a password is provided?
> 
> I want the password to be dynamic, combining it with the time or date. For example, entering the current time should execute the program.
> 
> I’m not sure, but I think this approach would be better than using encryption.
> 
> **In simple terms,** Instead of encryption, implement regular password protection. The password should change every minute, based on the current time.
> 
> **Example:** If the time is 12:00:
> 
> ```python
> from package_name.models import display
> display('filename', 1200)
> ```
> 
> This should execute the feature, with the time serving as the password.
> 
> For this issue, you can focus on the `models.py` feature only. We’ll create two additional issues for implementing the other features, as they have different functions—one copies to the clipboard and the other writes the entire file to the root directory.

Can you assign this issue to me

**Comment by this-is-yaash** [comment]
@sid7219, hey! Thank you for showing interest in this project. However, @Varun-malviya5 has already requested to work on this issue. Let's see if they're interested in taking it on first. If not, I’ll assign the issue to you!

**Comment by this-is-yaash** [comment]
@sid7219, I’ll be adding more issues soon, so please keep an eye on the issues section. If something interests you, just leave a comment, and I’ll assign it to you!

**Comment by Varun-malviya5** [comment]
yes i can try to do it without encryption and make use of models.py

**Comment by sajal-11111** [comment]
@this-is-yaash could you assign this issue to me ?

**Comment by this-is-yaash** [comment]
@Varun-malviya5 assigning this issue to you!

@sajal-11111 keep an eye on the **Issues** section, raise/request for the issues you want to work on!

**Comment by Varun-malviya5** [comment]
it will be completed by EOD

**Comment by this-is-yaash** [comment]
@Varun-malviya5 would you like to share your progression?
maybe we can assist you!

**Comment by Varun-malviya5** [comment]
i have changed the models.py function and have not changed anything in graph and piechart module but when i ahve created a pull request the other files are chjanged automatically and i cant find the original mone for graph and piechart , u may check the models.py ,i have resolved issues there

**Comment by this-is-yaash** [comment]
@Varun-malviya5 I noticed you removed your issue, are you no longer interested?

**Comment by Probatio-Diabolica** [comment]
Hey Yash! Please assign this to me thanks!!

**Comment by this-is-yaash** [comment]
@Probatio-Diabolica thanks for showing interest!
I want you to add password feature into `models.py` & `piechart.py`.

- [Check out this `graph.py` feature](https://github.com/AmateursLeague/sneaky-package/blob/main/package/graph.py), it already has implemented the password feature.
- I just want the password feature to be implemented in the `models.py` & `piechart.py`.

Let me know your thoughts about this, then I'll assign to you!

> ensure to go through the project and understood what it does!

**Comment by Probatio-Diabolica** [comment]
Thanks @this-is-yaash for giving the valuable hint. I see, the password works in this manner:
> takes password in the HHMM (time format, although i don't get it why time specific? Is the password not given by user but rather a pipeline from another file?)
> then it checks the current time with the password if the password is correct then it allows the module to be accessed which is in the stash module
> I think you want it to work like the "import someModule" .
Yeah It is doable, Can I please get to know the deadlines as well? So that I can manage my university mid terms and this project. 
Thanks for your time , good evening

**Comment by Wahid7852** [comment]
since it is a level 1 issue, a 24 hour time limit is reasonable, we can extent it by a bit if genuine reasons but not more than 48 hours

**Comment by this-is-yaash** [comment]
I assume you understood the requirement!
But again, the goal is to integrate the password feature into the remaining features, that's it!

Deadline is tomorrow EOD!
I'll assign this issue to you now!

**Comment by this-is-yaash** [comment]
make sure you use the development branch!(easy tracking)

**Comment by Probatio-Diabolica** [comment]
Thanks I'll begin my work on it.

**Comment by Wahid7852** [comment]
up for grab for everybody!

**Comment by Wahid7852** [comment]
how will you be tackling this issue @NikhilJ2005

**Comment by Wahid7852** [comment]
make sure that you're not using GPT generated contents or we will not consider you for the issue

**Comment by sociallyencrypted** [comment]
Can I please be assigned to this issue? Contributing through both GSSoc'24 Ext and Hacktober right now

**Comment by Wahid7852** [comment]
How will you tackle this issue @sociallyencrypted

**Comment by sociallyencrypted** [comment]
> How will you tackle this issue @sociallyencrypted

So, I plan on using passwords by hashing the password SHA256 using, using the pycrypto library and then encrypt the file using the hashed password. I'll be adding a new function to encrypt and store the file (this can again be another "covert" function like `show()`), and then I'll extend the `display()` method to have another argument to take a password. I'll also add error handling: if display() is called on an encrypted file without the password or with a wrong password, the user will be alerted about the same. this can be checked using a digital signature appended at the end of the encrypted file.

**Comment by sociallyencrypted** [comment]
Also, just saw the discussions about using the current time as the password. I assume that should be easier, as it has already been implemented once. Let me know which approach you want me to take

**Comment by this-is-yaash** [comment]
fixed #110
